zeebe/engine/src/main/java/io/camunda/zeebe/engine/processing/deployment/model/validation/ProcessConditionalStartEventValidator.java

```diff
@@ -0,0 +1,100 @@
+/*
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH under
+ * one or more contributor license agreements. See the NOTICE file distributed
+ * with this work for additional information regarding copyright ownership.
+ * Licensed under the Camunda License 1.0. You may not use this file
+ * except in compliance with the Camunda License 1.0.
+ */
+package io.camunda.zeebe.engine.processing.deployment.model.validation;
+
+import io.camunda.zeebe.model.bpmn.instance.Condition;
+import io.camunda.zeebe.model.bpmn.instance.ConditionalEventDefinition;
+import io.camunda.zeebe.model.bpmn.instance.Process;
+import io.camunda.zeebe.model.bpmn.instance.StartEvent;
+import io.camunda.zeebe.model.bpmn.instance.SubProcess;
+import java.util.Collection;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Set;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+import org.camunda.bpm.model.xml.validation.ModelElementValidator;
+import org.camunda.bpm.model.xml.validation.ValidationResultCollector;
+
+public class ProcessConditionalStartEventValidator implements ModelElementValidator<Process> {
+
+  @Override
+  public Class<Process> getElementType() {
+    return Process.class;
+  }
+
+  @Override
+  public void validate(
+      final Process process, final ValidationResultCollector validationResultCollector) {
+
+    // root-level start events
+    validateConditionalStartEventsDuplicate(
+        process.getChildElementsByType(StartEvent.class),
+        process,
+        validationResultCollector,
+        "Duplicate condition expression '%s' found in conditional start events of process '%s'. "
+            + "Condition expressions for conditional start events must be unique within a process.");
+
+    // event subprocess start events
+    final Collection<StartEvent> eventSubprocessStartEvents =
+        process.getChildElementsByType(SubProcess.class).stream()
+            .map(sub -> sub.getChildElementsByType(StartEvent.class))
```

Comment:
:wrench: @berkaycanbc I think we should only need event sub-processes, please exclude embedded sub-processes.

```
process.getChildElementsByType(SubProcess.class).stream()
            .filter(SubProcess::triggeredByEvent) // filter event sub-processes
            .map(sub -> sub.getChildElementsByType(StartEvent.class))
            .flatMap(Collection::stream)
            .toList();
```